### PR TITLE
for background tasks, uniquely identify them by background session id…

### PR DESCRIPTION
**Pull request by thuyBox**
… and session task id

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @thuyBox has signed the CLA. Thanks for the pull request!

File: BoxContentSDK/BoxContentSDK/BOXURLSessionManager.m [hunk 1]
```diff
@@ -28,17 +28,23 @@ @interface BOXURLSessionManager() <NSURLSessionDelegate, NSURLSessionTaskDelegat
 //A task delegate can always be re-associated back with a session task by calling associateSessionTaskId:withTaskDelegate:
 @property (nonatomic, strong, readwrite) id<BOXURLSessionManagerDelegate> defaultDelegate;
 
-//a map to associate a session task with its task delegate
+//a map to associate a progress session' task to its delegate
 //during session/task's delegate callbacks, we call appropriate methods on task delegate
-@property (nonatomic, readonly, strong) NSMapTable<NSNumber *, id<BOXURLSessionTaskDelegate>> *sessionTaskIdToTaskDelegate;
+@property (nonatomic, readonly, strong) NSMapTable<NSNumber *, id<BOXURLSessionTaskDelegate>> *progressSessionTaskIdToTaskDelegate;
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/BOXURLSessionManager.m:33]: Minor: name ...ToTaskDelegateMap ?

File: BoxContentSDK/BoxContentSDK/BOXURLSessionManager.m [hunk 2]
```diff
@@ -28,17 +28,23 @@ @interface BOXURLSessionManager() <NSURLSessionDelegate, NSURLSessionTaskDelegat
...
+
+//a map to associate a background session to its session task and session task delegate
+//there can be more than one background session at a time as an app takes over background sessions created from app extensions
+//during session/task's delegate callbacks, we call appropriate methods on task delegate
+@property (nonatomic, readonly, strong) NSMutableDictionary<NSString *, NSMapTable<NSNumber *, id<BOXURLSessionTaskDelegate>> *> *backgroundSessionIdToSessionTaskIdAndTaskDelegate;
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/BOXURLSessionManager.m:38]: same

**Review by boxtcarpel — COMMENTED** [review]
+1

File: BoxContentSDK/BoxContentSDK/BOXURLSessionManager.h [hunk 3]
```diff
@@ -130,14 +130,9 @@ totalBytesExpectedToSend:(int64_t)totalBytesExpectedToSend;
...
- * Associate a session task with its task delegate to handle callbacks for it, taskDelegate is not retained
+ * Associate a background session task with its task delegate to handle callbacks for it, taskDelegate is not retained
  */
-- (void)associateSessionTaskId:(NSUInteger)sessionTaskId withTaskDelegate:(id <BOXURLSessionTaskDelegate> )taskDelegate;
-
-/**
- * Dessociate a session task with its task delegate so the task delegate will no longer handle callbacks for the task
- */
-- (void)deassociateSessionTaskId:(NSUInteger)sessionTaskId;
+- (void)associateBackgroundSessionTaskId:(NSUInteger)sessionTaskId withTaskDelegate:(id <BOXURLSessionTaskDelegate> )taskDelegate;
```
Comment (irvingruan) [inline BoxContentSDK/BoxContentSDK/BOXURLSessionManager.h:135]: Minor: extra space after *Delegate>

**Comment by irvingruan** [comment]
+2

**Pull request merged**